org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsi/participants/XSIFormatterParticipant.java

```diff
@@ -155,4 +158,58 @@ public String getCurrentLineIndent(XMLBuilder xml, XMLFormattingOptions formatti
 		return indent.toString();
 	}
 
+	@Override
+	public boolean formatAttributeValueNew(XMLFormatterDocumentNew formatterDocument,
+			int indentLevel, XMLFormattingOptions formattingOptions, DOMAttr attr,
+			List<TextEdit> edits) {
+
+		String name = attr.getName();
+		String split = formattingOptions.getXsiSchemaLocationSplit();
+		int start = attr.getNodeAttrValue().getStart();
+		int tabSize = formattingOptions.getTabSize();
+		String attrValue = attr.getOriginalValue();
+
+		if (split.equals("none") || !XSISchemaModel.isXSISchemaLocationAttr(name, attr)) {
+			return true;
+		}
+
+		// Remove extra spaces between start of attribute value quote and actual value
+		formatterDocument.removeLeftSpaces(start + 1, // <foo attr = "|"
+		// <foo attr = " |value"
+		start + attr.getFirstContentOffset(attr.getOriginalValue()), edits);
+
+		for (int i = 0; i < attrValue.length(); i++) {
+			if (Character.isWhitespace(attrValue.charAt(i)) && !Character.isWhitespace(attrValue.charAt(i + 1))) {
+				int from = formatterDocument.getLeftWhitespacesOffset(start, start + i + 1);
+				//Insert newline and indent where required based on setting
+				if ((split.equals("onElement") && !StringUtils.isQuote(attrValue.charAt(from - start)))

```

Comment:
Do condition if (!StringUtils.isQuote(attrValue.charAt(from - start))) in a top level if